dish_grpc_influx2.py

```diff
@@ -0,0 +1,330 @@
+    fields = {"status": {}, "ping_stats": {}, "usage": {}}
+
+    def cb_add_item(key, val, category):
+        fields[category][key] = val
+
+    def cb_add_sequence(key, val, category, start):
+        for i, subval in enumerate(val, start=start):
+            fields[category]["{0}_{1}".format(key, i)] = subval
+
+    def cb_add_bulk(bulk, count, timestamp, counter):
+        if gstate.start_timestamp is None:
+            gstate.start_timestamp = timestamp
+            gstate.start_counter = counter
+        points = gstate.points if gstate.timebase_synced else gstate.deferred_points
+        for i in range(count):
+            timestamp += 1
+            points.append({
+                "measurement": BULK_MEASUREMENT,
+                "tags": {
+                    "id": gstate.dish_id
+                },
+                "time": timestamp,
+                "fields": {key: val[i] for key, val in bulk.items() if val[i] is not None},
+            })
+        if points:
+            # save off counter value for script restart
+            points[-1]["fields"]["counter"] = counter + count
+
+    now = time.time()
+    # work with UTC here
+    # now = time.mktime(datetime.utcnow().timetuple())
+    rc = dish_common.get_data(opts, gstate, cb_add_item, cb_add_sequence, add_bulk=cb_add_bulk)
+    if rc:
+        return rc
+
+    for category in fields:
+        if fields[category]:
+            gstate.points.append({
+                "measurement": "spacex.starlink.user_terminal." + category,
+                "tags": {
+                    "id": gstate.dish_id
+                },
+                "time": int(now),
+                "fields": fields[category],
+            })
+
+    # This is here and not before the points being processed because if the
+    # query previously failed, there will be points that were processed in
+    # a prior loop. This avoids having to handle that as a special case.
+    if opts.bulk_mode and not gstate.timebase_synced:
+        sync_timebase(opts, gstate)
+
+    if opts.verbose:
+        print("Data points queued: " + str(len(gstate.points)))
+
+    if len(gstate.points) >= FLUSH_LIMIT:
+        return flush_points(opts, gstate)
+
+    return 0
+
+
+def main():
+    opts = parse_args()
+
+    logging.basicConfig(format="%(levelname)s: %(message)s")
+
+    gstate = dish_common.GlobalState(target=opts.target)
+    gstate.points = []
+    gstate.deferred_points = []
+    gstate.timebase_synced = opts.skip_query
+    gstate.start_timestamp = None
+    gstate.start_counter = None
+
+    if "verify_ssl" in opts.icargs and not opts.icargs["verify_ssl"]:
+        # user has explicitly said be insecure, so don't warn about it
+        warnings.filterwarnings("ignore", message="Unverified HTTPS request")
+
+    signal.signal(signal.SIGTERM, handle_sigterm)
+    try:
+        gstate.influx_client = InfluxDBClient(**opts.icargs)
+    except TypeError as _err:
+        print('Error while creating influx client: ' + str(_err))
```

Comment:
This seems like it would continue while leaving gstate.influx_client unset.

It really should either sys.exit or re-throw the exception. Or just remove the try clause and let the exception abort the script run. The InfluxDB 1.x script only caught TypeError here in an effort to hack around old versions of the influxdb module that didn't support another hack-around which wouldn't be relevant when using the InfluxDB 2.x client module.